src/posix/platform/settings.cpp

```diff
@@ -323,17 +315,17 @@ otError otPlatSettingsDelete(otInstance *aInstance, uint16_t aKey, int aIndex)
 
         rval = write(swapFd, &key, sizeof(key));
         assert(rval == sizeof(key));
-        VerifyOrDie(rval == sizeof(key));
+        VerifyOrDie(rval == sizeof(key), OT_EXIT_ERROR_ERRNO);
 
         rval = write(swapFd, &length, sizeof(length));
         assert(rval == sizeof(length));
-        VerifyOrDie(rval == sizeof(length));
+        VerifyOrDie(rval == sizeof(length), OT_EXIT_ERROR_ERRNO);
 
         swapWrite(swapFd, length);
     }
 
 exit:
-    VerifyOrDie(error != OT_ERROR_PARSE);
+    VerifyOrDie(error != OT_ERROR_PARSE, OT_EXIT_ERROR_ERRNO);
```

Comment:
`errno` is not guaranteed to be set here.

Reply:
updated